### PR TITLE
test: add tests for 'get_job_list'

### PR DESCRIPTION
**Pull request by danielboloc**
# Overview

Test the function `get_job_list` from `cloudos.clos`

Uses the `mock` and `responses` python packages to mock HTTP GET requests.

The test has the following functions each testing a different part:

- `test_get_job_list_correct_response` -> test if the function returns `requests.models.Response` and the `status_code` of the response is `200`

- `test_get_job_list_incorrect_response` -> test if the function raises the error `BadRequestException`

# Running the test

```console
python -m pytest -v tests/test_get_job_list.py
```

The response should be that all 2 tests have passed:

image

### CODE REVIEW

File: tests/test_get_job_list.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            url=f"{CLOUDOS_URL}/api/v1/jobs?{search_str}",
+            body=error_json,
+            headers=header,
+            match=[matchers.query_param_matcher(params)],
+            status=400)
+    # raise 400 error
+    with pytest.raises(BadRequestException) as error:
+        # check if it failed
+        clos = Cloudos(apikey=APIKEY, cromwell_token=None, cloudos_url=CLOUDOS_URL)
+        response = clos.get_job_list(WORKSPACE_ID)
```
Comment (dapineyro) [inline tests/test_get_job_list.py:75]:
```suggestion
        clos.get_job_list(WORKSPACE_ID)
```
`response` is never used, so we don't need to create it

**Review by dapineyro — CHANGES_REQUESTED** [review]
Awesome work @danielboloc ! I left some style suggestions only. Please, merge the latests changes from `dev` to your branch and move this file `tests/test_get_job_list.py` into the new folder -> `test/test_clos/test_get_job_list.py`

File: tests/test_get_job_list.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    responses.add(
+            responses.GET,
+            url=f"{CLOUDOS_URL}/api/v1/jobs?{search_str}",
+            body=create_json,
+            headers=header,
+            match=[matchers.query_param_matcher(params)],
+            status=200)
+    # start cloudOS service 
+    clos = Cloudos(apikey=APIKEY, cromwell_token=None, cloudos_url=CLOUDOS_URL)
+    # get mock response
```
[the remark this answers is not in the document]
Comment (danielboloc) [reply tests/test_get_job_list.py:41]: Is supposed to be whitespace, I have noticed this in my VSC installation that sometimes it put a yellow box when I paste a phrase with spaces

File: tests/test_get_job_list.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    response = clos.get_job_list(WORKSPACE_ID)
+    # check the response
```
[the remark this answers is not in the document]
Comment (danielboloc) [reply tests/test_get_job_list.py:43]: Is supposed to be whitespace, I have noticed this in my VSC installation that sometimes it put a yellow box when I paste a phrase with spaces

File: tests/test_get_job_list.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    assert response.status_code == 200
+    assert isinstance(response, requests.models.Response)
+
+@mock.patch('cloudos.clos', mock.MagicMock())
+@responses.activate
+def test_get_job_list_incorrect_response():
+    """
+    Test 'get_job_list' to fail with '400' response
+    """
+    # prepare error message
```
[the remark this answers is not in the document]
Comment (danielboloc) [reply tests/test_get_job_list.py:53]: Is supposed to be whitespace, I have noticed this in my VSC installation that sometimes it put a yellow box when I paste a phrase with spaces

File: tests/test_get_job_list.py (hunk 1 above)
Comment (danielboloc) [reply tests/test_get_job_list.py:75]: true

File: tests/test_clos/test_get_job_list.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    search_str = f"teamId={WORKSPACE_ID}&apikey={APIKEY}"
+    # mock GET method with the .json
+    responses.add(
+            responses.GET,
+            url=f"{CLOUDOS_URL}/api/v1/jobs?{search_str}",
+            body=error_json,
+            headers=header,
+            match=[matchers.query_param_matcher(params)],
+            status=400)
+    # raise 400 error
```
Comment (dapineyro) [inline tests/test_clos/test_get_job_list.py:72]:
```suggestion
    # raise 400 error
```
Somehow, this is still a different symbol in my system and also in Github

**Review by dapineyro — CHANGES_REQUESTED** [review]
Thanks for addressing my comments @danielboloc . One of those weird whitespace characters are still there. Otherwise, the code is now great! :100:

File: tests/test_clos/test_get_job_list.py (hunk 5 above)
Comment (danielboloc) [reply tests/test_clos/test_get_job_list.py:72]: hmm could be because I use Mac, I will look it up

**Review by dapineyro — APPROVED** [review]
Looks great! Approved! :+1:

**Pull request merged**